src/sentry/workflow_engine/models/detector.py

```diff
@@ -4,34 +4,46 @@
 import logging
 from typing import TYPE_CHECKING, Any
 
+from django.conf import settings
 from django.db import models
 from django.db.models import UniqueConstraint
 
 from sentry.backup.scopes import RelocationScope
 from sentry.db.models import DefaultFieldsModel, FlexibleForeignKey, region_silo_model
+from sentry.db.models.fields.hybrid_cloud_foreign_key import HybridCloudForeignKey
 from sentry.issues import grouptype
 from sentry.issues.grouptype import GroupType
 from sentry.models.owner_base import OwnerModel
 
+from .json_config import JSONConfigBase
+
 if TYPE_CHECKING:
     from sentry.workflow_engine.processors.detector import DetectorHandler
 
 logger = logging.getLogger(__name__)
 
 
 @region_silo_model
-class Detector(DefaultFieldsModel, OwnerModel):
+class Detector(DefaultFieldsModel, OwnerModel, JSONConfigBase):
     __relocation_scope__ = RelocationScope.Organization
 
-    organization = FlexibleForeignKey("sentry.Organization")
+    # TODO - Finish removing this field
+    organization = FlexibleForeignKey("sentry.Organization", on_delete=models.CASCADE, null=True)
+
+    project = FlexibleForeignKey("sentry.Project", on_delete=models.CASCADE, null=True)
     name = models.CharField(max_length=200)
 
     # The data sources that the detector is watching
     data_sources = models.ManyToManyField(
         "workflow_engine.DataSource", through="workflow_engine.DataSourceDetector"
     )
 
-    # The conditions that must be met for the detector to be considered 'active'
+    # If the detector is not enabled, it will not be evaluated. This is how we "snooze" a detector
+    enabled = models.BooleanField(default=True)
+
+    # Optionally set a description of the detector, this will be used in notifications
+    description = models.TextField(blank=True, null=True)
```

Comment:
Blank means empty string, null means can be null. Do you need both of these set?